src/textobject/textobject.ts

```diff
@@ -971,3 +972,56 @@ export class SelectAroundArgument extends SelectArgument {
   keys = ['a', 'a'];
   override selectAround = true;
 }
+
+/**
+ * This class creates text objects based on the syntax symbols generated by VsCode's LSP.
+ * This allows to easily select class objects or functions for example by inheriting it
+ * and creating a whitelist.
+ */
+abstract class SelectASymbol extends TextObject {
+  // The whitelist is the filter used to only select the Symbol type that we want
+  abstract whitelist: Set<SymbolKind>;
+
+  public override async execAction(position: Position, vimState: VimState): Promise<IMovement> {
+    let start: Position = vimState.cursorStartPosition;
+    let stop: Position = vimState.cursorStopPosition;
+
+    const symbols = await TextEditor.getSymbols(vimState.editor);
```

Comment:
Do you think it would be worth it to cache these symbols for a given `TextDocument.version`?